ReactNativeClient/lib/services/filterParser.ts

```diff
@@ -0,0 +1,62 @@
+/* eslint-disable no-unused-vars */
+/* eslint-disable no-cond-assign */
+
+// match[1] AND/OR tagName:tagValue
+// match[2] AND/OR
+// match[3] -
+// match[4] tagName
+// match[6] tagValue
+// const filterRegex = /(\b(AND|OR)\s+)?(-?)([\w]+|"[^"]+")\s*(:\s*([^\s:*"]+|"[^"]+"))?/giu;
+const filterRegex = /(\b(AND|OR)\s+)?(-?)(([\w]+|"[^"]+")\s*(:\s*([^\s:*"]+|"[^"]+"))|(\b[^:\s"]+\b|"[^"]+"))/giu;
```

Comment:
I'm afraid this kind of regex breaks easily and, additionally, is very hard to make sense of. Wouldn't it be possible to build an actual query parser like it's done in SearchEngine::parseQuery?

Reply:
I'm torn on this.

I completely agree with all your points. In fact, I fixed a bug in Trilium's filterRegex when I was studying their code, thus demonstrating their breakability. And I get that it's hard to make sense of initially and so hard to maintain(and modify).

The points against are that it's convenient and with adequate documentation quite easy to understand. 
That regex was designed for situations like this and that the queryParser might get long and complicated once we get into smart values and dates etc. (though that could be argued both ways). Also, I might be (over)confident that I got this right.

What I'll do is try to make the queryParser work and only resort to regex as a backup if all else fails. Is that OK?